### PR TITLE
Primary charity admin

### PR DESCRIPTION
**Pull request by karinamzalez**
test for admin demotion still needs to be written but it's functional.

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -122,7 +122,7 @@ def create_users
     100.times do
       user = User.create!(
                             username:  Faker::Internet.user_name,
-                            password:  Faker::Internet.password,
+                            password_digest:  Faker::Team.state,
                             cell:      1112223333,
                             email:     Faker::Internet.email,
```
Comment (lucyfox4131) [inline db/seeds.rb:129]: Why was this changed to Team? I think for clarity we should keep this as internet password

File: db/seeds.rb [hunk 2]
```diff
@@ -140,11 +140,11 @@ def create_roles
   end
 
   def create_admin_user_roles
-    30.times do
-      user = User.find(rand(1..100))
-      name = ["platform_admin", "charity_admin", "charity_original_admin", "registered_user"].sample
-      role = Role.find_by(name: name)
-      user.roles << role
+      User.all.each do |user|
+        name = ["platform_admin", "charity_admin", "charity_original_admin", "registered_user"].sample
```
Comment (lucyfox4131) [inline db/seeds.rb:159]: Let's make sure we change the name of this. It's not charity original admin in the project anymore nor are we creating a registered user role as we are only doing roles for admins. We should probably change the seed file to reflect this.

File: app/controllers/admin/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      else
+        redirect_to dashboard_path
+      end
+    else
+      flash.now[:warning] = @user.errors.full_messages.join(", ")
+      render :edit
+    end
+  end
+
+  def destroy
```
Comment (AnnaCW) [inline app/controllers/admin/users_controller.rb:41]: This shouldn't be a destroy action in the controller, since it is just changing status of a user, not deleting a user from the database.

File: app/assets/stylesheets/application.css [hunk 4]
```diff
@@ -88,6 +90,11 @@
   color: white;
 }
 
+.white-texxt {
+  font-size: 22px;
+  color: white;
+}
+
```
Comment (lucyfox4131) [inline app/assets/stylesheets/application.css:97]: I don't think the name of this css rule should have two x's in it. Makes it more challenging to remember what it is called and to reuse it.

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -9,7 +9,7 @@ def create
     if user && user.authenticate(params[:session][:password])
       session[:user_id] = user.id
       flash[:success] = "Hi, #{user.username}!"
-      if user.charity_admin? || user.platform_admin?
+      if user.charity_admin? || user.platform_admin? || user.charity_original_admin?
```
Comment (AnnaCW) [inline app/controllers/sessions_controller.rb:12]: Use admin? method here.

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -39,16 +43,16 @@ def update
...
-    if (!current_user || current_user.id != params[:id].to_i)
-      flash[:warning] = "Oops, you visited the wrong page."
-      redirect_to root_path
-    end
-  end
+   def check_for_correct_user
+     if !current_user || current_user.id != params[:id].to_i
+       flash[:warning] = "Oops, you visited the wrong page."
+       redirect_to root_path
+     end
```
Comment (lucyfox4131) [inline app/controllers/users_controller.rb:56]: We can get rid of this method. It's not necessary, there are better ways of ensuring a user is only able to see their own information as opposed to someone else's.

File: app/models/user.rb [hunk 7]
```diff
@@ -32,12 +33,32 @@ def charity_admin?
...
+  end
+
+  def self.charity_admins(current_user_charity)
+    User.all.map do |user|
+      if !user.charities.empty? && user.charities.first == current_user_charity
+        if !user.roles.empty? && user.roles.first.name == "charity_admin"
+          user
+        end
+      end
+    end.compact
```
Comment (lucyfox4131) [inline app/models/user.rb:61]: This looks messy. Let's look into refactoring this.

File: app/views/admin/dashboards/show.html.erb [hunk 8]
```diff
@@ -22,3 +22,52 @@
...
+          </div>
+
+          <div class="form-group">
+            <%= f.label :cell %>
+            <%= f.telephone_field :cell, class: "form-control" %>
+          </div>
+
+        <%= f.submit "Update Admin", class: "btn btn-info active" %>
+        <% end %>
+        <%= button_to "Remove Admin Status", admin_user_path(user), method: :delete, class: "btn btn-warning active" %>
```
Comment (AnnaCW) [inline app/views/admin/dashboards/show.html.erb:69]: We just want to change user status, not delete.

File: app/controllers/admin/users_controller.rb (hunk 3 above)
Comment (lucyfox4131) [inline app/controllers/admin/users_controller.rb:41]: I agree with Anna. I think this is really an update. The action itself may not even need to be routed through a controller. We can maybe either make a poro or an action on the model that handles the status update here.

File: app/views/layouts/application.html.erb [hunk 9]
```diff
@@ -37,10 +37,10 @@
 
       </ul>
       <ul class="nav navbar-nav navbar-right">
-        <% if current_user && current_user.charity_admin? %>
-          <li class="custom-nav"> <%= link_to "Logged in as #{current_user.username}", admin_dashboard_path %> </li>
+        <% if current_user && (current_user.charity_original_admin? || current_user.charity_admin?) %>
```
Comment (lucyfox4131) [inline app/views/layouts/application.html.erb:40]: This looks like a very messy conditional to have in the view. Let's make sure we refactor this. We can push this logic down into the model.

File: app/views/admin/dashboards/show.html.erb [hunk 10]
```diff
@@ -22,3 +22,52 @@
...
+    <label>Filter By Username  </label>
+    <input type="text" placeholder="Username" id="user_filter_username">
+</div>
+
+<div class="box">
+  <%= link_to "Create New Admin", new_admin_user_path, class: "btn button-indigo" %>
+</div>
+
+<div class="col-sm-12 col-sm-offset-1 users">
+  <% @users.each do |user| %>
```
Comment (lucyfox4131) [inline app/views/admin/dashboards/show.html.erb:45]: As we are dealing with specifically admins here we should make sure that we change this variable name to admin as opposed to user. It will be more clear to someone reading the code later that we are specifically displaying only admins in this section of the page.
